src/district0x/ui/db.cljs

```diff
@@ -26,6 +27,9 @@
                      :ids-by-form {}
                      :settings {:from-active-address-only? false}}})
 
-
+(defn try-resolving-address [db addr]
+  (if-not (web3/address? addr)
+    (get-in db [:ens/records (namehash addr) :ens.record/address] "0x")
```

Comment:
Don't forget to add spec for it into `name-bazaar.ui.spec`

Reply:
For data storage structure, it's benefitial not to burden it with source's, or consumer's schemas, i.e. keep it as flat as possible, so that the customer subs could build their queries in the uniformed way (i'll keep tooling for solving this out of the scope :) ). If the subscription just wants to know the address of the node, why should it care about resolvers at all?

Reply:
But i'm not plannig going against the grain here, will do it the way you described. Just walked these rakes before.

Reply:
Well, I think it's okay for subscriptions to deal with internal db structure in flat state. It's the consumer of subscriptions (components) that shouldn't care. So subscription should serve just the address of the node, but internally deal with resolvers.